### PR TITLE
Restrict symbol resolving to language

### PR DESCRIPTION
**Pull request by oxisto**
Currently, the symbol resolver runs wild and considers all nodes of all languages. When we have more instances where we have multiple languages in the same component, this craates a problem.

This PR now only considers the same language as the ref we are looking for.

### CODE REVIEW

File: cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/ScopeManager.kt [hunk 1]
```diff
@@ -900,6 +902,18 @@ class ScopeManager : ScopeProvider {
     override val scope: Scope?
         get() = currentScope
 
+    /**
+     * A convenience function to call [lookupSymbolByName] with the properties of [node]. The
+     * arguments [scope] and [predicate] are forwarded.
+     */
+    fun lookupSymbolByNameOfNode(
+        node: HasNameAndLocation,
+        scope: Scope? = currentScope,
```
Comment (oxisto) [inline cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/ScopeManager.kt:911]:
```suggestion
        scope: Scope? = node.scope,
```

**Comment by oxisto** [comment]
Looking for review ;) @maximiliankaul @KuechA @konradweiss @lshala

File: cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt [hunk 2]
```diff
@@ -590,7 +590,11 @@ open class SymbolResolver(ctx: TranslationContext) : ComponentPass(ctx) {
         var candidates = mutableSetOf<Declaration>()
         val records = possibleContainingTypes.mapNotNull { it.root.recordDeclaration }.toSet()
         for (record in records) {
-            candidates.addAll(ctx.scopeManager.lookupSymbolByName(record.name.fqn(symbol)))
+            candidates.addAll(
+                ctx.scopeManager.lookupSymbolByName(record.name.fqn(symbol), record.language) {
+                    it.language == record.language
```
Comment (KuechA) [inline cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt:595]: Is this additional predicate really required? Shouldn't this already be ensured by the `languageOnly` checks?

File: cpg-language-go/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/GoExtraPass.kt [hunk 3]
```diff
@@ -389,8 +389,10 @@ class GoExtraPass(ctx: TranslationContext) : ComponentPass(ctx) {
 
         // Try to see if we already know about this namespace somehow
         val namespace =
-            scopeManager.lookupSymbolByName(import.name, null).filter {
-                it is NamespaceDeclaration && it.path == import.importURL
+            scopeManager.lookupSymbolByNameOfNode(import).filter {
+                it is NamespaceDeclaration &&
+                    it.path == import.importURL &&
+                    it.language == import.language
```
Comment (KuechA) [inline cpg-language-go/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/GoExtraPass.kt:395]: Again, there's this additional language check in the predicate

**Review by KuechA — APPROVED** [review]

File: cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt (hunk 2 above)
Comment (oxisto) [reply cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/SymbolResolver.kt:595]: Ah this is a leftover before I introduced the additional argument, will remove these

**Pull request merged**